File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/Location.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Entity
@Table(name = "locations", 
       uniqueConstraints = @UniqueConstraint(columnNames = {"state", "district", "pincode"}))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Location {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank(message = "Country is required")
    @Size(max = 100, message = "Country name cannot exceed 100 characters")
    @Column(nullable = false, length = 100)
    private String country;
    
    @NotBlank(message = "State is required")
    @Size(max = 100, message = "State name cannot exceed 100 characters")
    @Column(nullable = false, length = 100)
    private String state;
    
    @NotBlank(message = "District is required")
    @Size(max = 100, message = "District name cannot exceed 100 characters")
    @Column(nullable = false, length = 100)
    private String district;
    
    @NotBlank(message = "Pincode is required")
    @Pattern(regexp = "^[1-9][0-9]{5}$", message = "Invalid pincode format")
    @Column(nullable = false, length = 6)
    private String pincode;
    
    @Pattern(regexp = "^(ACTIVE|INACTIVE)$", message = "Status must be ACTIVE or INACTIVE")
    @Builder.Default
    @Column(nullable = false)
    private String status = "ACTIVE";
}

File: LifeSamadhan/Backend/Models/Location.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("locations")]
    public class Location
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Required(ErrorMessage = "Country is required")]
        [StringLength(50, ErrorMessage = "Country name is too long")]
        [Column("Country")]
        public string Country { get; set; } = "India";

        [Required(ErrorMessage = "State is required")]
        [StringLength(100, ErrorMessage = "State name is too long")]
        [Column("State")]
        public string State { get; set; } = string.Empty;

        [Required(ErrorMessage = "District is required")]
        [StringLength(100, ErrorMessage = "District name is too long")]
        [Column("District")]
        public string District { get; set; } = string.Empty;

        [Required(ErrorMessage = "Pincode is required")]
        [RegularExpression(@"^[1-9][0-9]{5}$",
            ErrorMessage = "Invalid Indian pincode")]
        [Column("Pincode")]
        public string Pincode { get; set; } = string.Empty;

        [RegularExpression("^(ACTIVE|INACTIVE)$",
            ErrorMessage = "Status must be ACTIVE or INACTIVE")]
        [Column("Status")]
        public string Status { get; set; } = "ACTIVE";
    }
}
